ansible/playbooks/roles/preflight/tasks/main.yml

```
@@ -30,6 +30,27 @@
 
 - include_tasks: check-os.yml
 
+- include_vars:
+    file: roles/common/vars/main.yml
+    name: common_vars
+
+- name: Validate network configuration
+  when:
+    - common_vars.provider == "any"
+    - common_vars.specification.cloud is undefined
+  block:
+  - name: Validate if ansible_default_ipv4.address matches address from inventory
+    when: ansible_default_ipv4 | length > 0
```

```suggestion
    when: ansible_default_ipv4.address is defined
```
I would use condition which tell you that this is dictionary since `length` is often used for lists.